integrations/sonarqube/client.py

```diff
@@ -288,13 +303,29 @@ async def get_all_projects(self) -> AsyncGenerator[list[dict[str, Any]], None]:
         :return (list[Any]): A list containing projects data for your organization.
         """
         logger.info(f"Fetching all projects in organization: {self.organization_id}")
-        self.metrics = cast(
-            SonarQubeProjectResourceConfig, event.resource_config
-        ).selector.metrics
-        components = await self.get_components()
-        for component in components:
-            project_data = await self.get_single_project(project=component)
-            yield [project_data]
+        selector = cast(SonarQubeProjectResourceConfig, event.resource_config).selector
+        self.metrics = selector.metrics
+
+        all_projects = {}
+        for project in await self.send_paginated_api_request(
+            endpoint=Endpoints.PROJECTS, data_key="components"
+        ):
+            project_key = project.get("key")
+            all_projects[project_key] = project
+
+        if selector.use_internal_api:
+            # usint the internal API to fetch more data
+            components = await self.get_components(Endpoints.PROJECTS_INTERNAL)
+            for component in components:
+                all_projects[component["key"]] = {
+                    **all_projects.get(component["key"], {}),
+                    **component,
+                }
+
+        yield [
+            await self.get_single_project(component)
+            for component in all_projects.values()
+        ]
```

Comment:
this can be very prune to errors if one fails get_single_project it will fail all projects